### ISSUE TITLE
Simulating arrow keys with shortcut without layers

### ISSUE THREAD

**Issue by spaceunifyfifty** [kmonad/kmonad#863]
I want `alt+jikl` to simulate arrow keys. I don't want to use a layer toggle for this, because that will make my `alt` key no longer function as `alt` which I use for other shortcuts on my system like `alt+tab` for window switching and many more. I also do not want to use two different keys or have `tap` vs `hold` behavior on alt. 

Does KMonad provide a way to make shortcuts like `alt+j` trigger arbitrary key simulation like `alt+j = left`? keyd apparently provides this functionality but I'd prefer to only use kmonad.

**Comment by jokesper** [comment]
No without layers this isn't possible, since combos (#157) are not implemented
What you could do instead is a layers.
Depending on what exactly you want there are different options:
- If you want the alt key only pressed when you press other keys you would need a full layer with the `around alt ...`
- If you want alt to be always pressed, you can have
`(around alt (layer-toggle ...))` and the nextmlayerwcould be full of fallthrough except for the arrow part
- If you want to release alt when pressing the arrow keys you can do the same but with an `(tap-macro-release (release-only alt) lft (press-only alt))` though this results in wierd behavior when pressing multiple "arrow keys".
In the last two options you would have a layer full of `_`
you could with the _next_ nelease use a named source block to only remap the jikl on your new layer and possibly make it more readable

**Comment by jokesper** [comment]
Yes, it is. I don't think there is a better solution currently available to you than prefixxing every button with `A-` in a new layer.

**Comment by jokesper** [comment]
> This?
> ```lisp
>   toggle-alt-layer (around lalt (layer-toggle my-layer))
> ```
No this would be the second solution.
I was meaning something more like this:
```closure
(defsrc
  q w e r t y
  j i k l
  alt spc
)
(deflayer default
  q w e r t y
  j i k l
  (layer-toggle with-alt) spc
)
(deflayer with-alt
  A-q A-w A-e A-r A-t A-y
  left down up right A-spc
)
```
A bit verbose but the best solution currently possible

**Comment by jokesper** [comment]
> This?
> ```lisp
>   toggle-alt-layer (around lalt (layer-toggle my-layer))
> ```
No this would be the second solution.
I was meaning something more like this:
```closure
(defsrc
  q w e r t y
  j i k l
  alt spc
)
(deflayer default
  q w e r t y
  j i k l
  (layer-toggle with-alt) spc
)
(deflayer with-alt
  A-q A-w A-e A-r A-t A-y
  left down up right A-spc
)
```
A bit verbose but the best solution currently possible.

**Comment by jokesper** [comment]
After having reread your question I think you would have a bit of difficulty with alt+tab.
Switching to the second or third last active window requires mulitple taps of tab while alt is being held.
This is probably possible with a hacky solution for switching, `press-only` and `release-only`
